storage/internal/elastic/config.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 Bull S.A.S. Atos Technologies - Bull, Rue Jean Jaures, B.P.68, 78340, Les Clayes-sous-Bois, France.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elastic
+
+import (
+	"github.com/spf13/cast"
+	"github.com/ystia/yorc/v4/config"
+	"github.com/ystia/yorc/v4/log"
+	"reflect"
+	"time"
+)
+
+// elasticStoreConf represents the elastic store configuration that can be set in store.properties configuration.
+type elasticStoreConf struct {
+	// The ES cluster urls (array or CSV)
+	esUrls []string `json:"es_urls"`
+	// The path to the CACert file when TLS is activated for ES
+	caCertPath string `json:"ca_cert_path"`
+	// All index used by yorc will be prefixed by this prefix
+	indicePrefix string `json:"index_prefix" default:"yorc_"`
+	// When querying logs and event, we wait this timeout before each request when it returns nothing
+	// (until something is returned or the waitTimeout is reached)
+	esQueryPeriod time.Duration `json:"es_query_period" default:"5s"`
+	// This timeout is used to wait for more than refresh_interval = 1s when querying logs and events indexes
+	esRefreshWaitTimeout time.Duration `json:"es_refresh_wait_timeout" default:"5s"`
+	// When querying ES, force refresh index before waiting for refresh
+	esForceRefresh bool `json:"es_force_refresh" default:"true"`
+	// This is the maximum size (in kB) of bulk request sent while migrating data
+	maxBulkSize int `json:"max_bulk_size" default:"4000"`
+	// This is the maximum size (in term of number of documents) of bulk request sent while migrating data
+	maxBulkCount int `json:"max_bulk_count" default:"1000"`
+}
+
+// Get the tag for this field (for internal usage only: fatal if not found !).
+func getElasticStorageConfigPropertyTag(fn string, tn string) string {
+	t := reflect.TypeOf(elasticStoreConf{})
```

Comment:
mapstructure can be used here instead of reflect (search mapstructure.Decode in Yorc code)

Reply:
Otherwise, it would be better if  reflect.TypeOf(elasticStoreConf{}) is done only once